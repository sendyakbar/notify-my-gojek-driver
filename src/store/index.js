import { createStore, combineReducers } from 'redux'

// reducers
import List from '../reducers/list'
import Search from '../reducers/search'

const reducers = combineReducers({
    List,
    Search
})

const store = createStore(reducers)

export default store