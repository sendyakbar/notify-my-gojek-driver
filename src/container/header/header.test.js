import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Header from './header.jsx';
import ConnectedHeader from './header.jsx'
import { Provider } from 'react-redux';
import store from '../../store'

describe('<Header> component', () => {
    const wrapper = shallow(<Header />)
    test('renders correctly', () => {
        expect(wrapper).toHaveLength(1)
    })
})
describe('<ConnectedHeader> redux-connected component', () => {
    test('renders correctly', () => {
        const connectedHeaderWrapper = shallow(
            <Provider store={store}>
                <ConnectedHeader />
            </Provider>
        )
        expect(connectedHeaderWrapper).toHaveLength(1)
    })
    test('renders correctly and matched snapshot', () => {
        const tree = renderer.create(
            <Provider store={store}>
                <ConnectedHeader />
            </Provider>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
