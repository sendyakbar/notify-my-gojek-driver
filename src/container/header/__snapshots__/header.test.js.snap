// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectedHeader> redux-connected component renders correctly and matched snapshot 1`] = `
<header
  className="App-header"
>
  <img
    alt="logo"
    className="App-logo"
    src="https://imgee.s3.amazonaws.com/imgee/638b26370a794db38dfef92fa2bfe60f.png"
  />
  <div>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="Type driver name..."
        type="text"
        value=""
      />
    </form>
  </div>
</header>
`;
